pkgs/by-name/cl/clamtk/package.nix

```diff
@@ -0,0 +1,69 @@
+{ lib
+, perlPackages
+, fetchFromGitHub
+, makeWrapper
+, gobject-introspection
+, perl
+, clamav
+}:
+let
+  clamavpath = lib.makeBinPath [ clamav ];
+in
+perlPackages.buildPerlPackage rec {
+  pname = "clamtk";
+  version = "6.16";
+
+  src = fetchFromGitHub {
+    owner = "dave-theunsub";
+    repo = "clamtk";
+    rev = "v${version}";
+    hash = "sha256-o6OaXOXLykTUuF/taKnEhZRV04/3nlU5aNY05ANr1Ko=";
+  };
+
+  nativeBuildInputs = [ makeWrapper gobject-introspection ];
+  buildInputs = [ perl clamav ];
+  propagatedBuildInputs = with perlPackages; [ Glib LWP LWPProtocolHttps TextCSV JSON LocaleGettext Gtk3 ];
+
+  preConfigure = "touch Makefile.PL";
+  doCheck = false;
+  outputs = [ "out" "man" ];
+
+  postPatch = ''
+    # Set the correct paths for clamav programs
+    sed -i lib/App.pm -e  "93s@.*@        : \'${clamavpath}/freshclam\';@"
```

Comment:
This is a bit iffy. I'd much prefer something like:
```sh
substituteInPlace lib/App.pm \
  --replace /usr/bin/freshclam ${lib.getBin clamav}/bin/freshclam \
  --replace /usr/bin/sigtool ${lib.getBin clamav}/bin/sigtool
```
and so on for the others

If you are dependent on specific line numbers, it gets really painful to maintain

Reply:
Thanks! Your method is much better. Have replaced `sed` with `substituteInPlace` for all occurrences in `App.pm` as suggested. Also for the icons path. And also managed to do it for `crontab`. Much appreciated as always. Many thanks!